cmake/CMakeLists.txt

```diff
@@ -181,6 +182,16 @@ if(onnxruntime_ENABLE_LTO)
     endif()
 endif()
 
+if(onnxruntime_DISABLE_RTTI)
+  if(MSVC)
+    # Disable RTTI and turn usage of dynamic_cast and typeid into errors
+    add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/GR->" "$<$<COMPILE_LANGUAGE:CXX>:/we4541>")
+    add_compile_definitions(ORT_NO_RTTI GOOGLE_PROTOBUF_NO_RTTI)
+  else()
+    message(SEND_ERROR "Disabling RTTI is currently only supported in MSVC")
+  endif()
```

Comment:
Why limit to MSVC vs using -fno-rtti for other compilers? Is there a build we have where that would not work?

Reply:
It's not a standard C++ feature (compliant C++ is expected to have RTTI) and support among compilers is not uniform (e.g. GCC allows dynamic_cast to void* and to an unambiguous upcast, which don't require RTTI, while MSVC will error at every usage of dynamic_cast). I didn't want to enable it globally without testing first, since other compilers might have quirks, and I only tested in Windows/MSVC.

Though if you're interested, I can enable it globally and run some pipelines to see if nothing breaks.

Reply:
I'd love to know if anything else breaks. We're trying to reduce binary size across all platforms so will want to enable this wherever possible. I can understand some things aren't uniform, but if you assume disabling rtti means zero usage of dynamic_cast and typeid shouldn't things work as expected as you're not relying on any compiler specific pseudo-rtti features?